src/elisp/treemacs-filewatch-mode.el

```diff
@@ -181,7 +181,8 @@ file from caches if it has been deleted instead of waiting for file processing."
             (treemacs-run-in-every-buffer
              (treemacs--on-rename old-name new-name))
             (treemacs--set-refresh-flags (treemacs--nearest-parent-directory old-name) 'deleted old-name)
-            (treemacs--set-refresh-flags (treemacs--nearest-parent-directory new-name) 'created new-name))
+            (when (--none? (funcall it (treemacs--filename new-name) new-name) treemacs-ignored-file-predicates)
```

Comment:
Strange rename event indeed. I would've thought that emacs creates that backup file instead of renaming things.

At any rate this sort of thing should be caught by `treemacs--is-event-relevant?`. You'll need to check for `(eq action 'renamed)` and then extract and test the new name.

A file's parent should be fetched with `treemacs--parent`. It looks like `treemacs--nearest-parent-directory` missed a refactoring round or two, and is basically useless. I'll slot it for removal.